### DISCUSSION TITLE
ApplicationData vs LocalApplicationData on Windows

### DISCUSSION THREAD

**Discussion by quvide** [ppy/osu-framework#6201] [Q&A] [answered]
I noticed this PR changed some logic around storage locations and added a comment: https://github.com/ppy/osu-framework/pull/6196/files#diff-47a19f45206aa7173d306c754da98cb703776dbcfc69c8f39c20dee404972794R32

```// The base implementation returns %LOCALAPPDATA%, but %APPDATA% is a better default on Windows.```

I see that there's no actual location change to the Windows implementation, but I'm wondering what the rationale is to consider `%APPDATA%` better than `%LOCALAPPDATA%`? Actual roaming profiles are perhaps not very common in gaming environments, but couldn't this potentially cause problems when storing large amounts of data in a domain with roaming user profiles enabled? Or am I missing something?

**Comment by peppy** [comment, marked as answer]
It's a sane default that works for most users (and is used this way by most applications for all user data).

If a user is actually replicating their roaming profile, they are more than welcome to change the folder being used. Or rather, if a developer is making an app using osu!framework where this would be a consideration, they can add support for this (like we've done with osu!).
